### PR TITLE
GH-2331 Query Editor Shows Done 0 results Though Results are available

### PR DESCRIPTION
**Pull request by Akhilsnaik**
## What changes were proposed in this pull request?

- GH-2331 - Query Editor Shows Done 0 results Though Results are available

## How was this patch tested?

- Tested Manually in UI, there is no Impact or issues, Tested all scenario's .

Screenshot 2021-06-09 at 9 56 01 AM
Screenshot 2021-06-09 at 9 55 04 AM

Please review [Hue Contributing Guide](https://github.com/cloudera/hue/blob/master/CONTRIBUTING.md) before opening a pull request.

### CODE REVIEW

**Comment by Akhilsnaik** [comment]
@JohanAhlen  Potential Issue faced by many when Results are fetched from impala on second API call , This covers the scenario of wrongly showing done 0 results for some time when the results are actually available.
tracked with `hasMore`  property, Please review and approve.

File: desktop/libs/notebook/src/notebook/templates/editor_components.mako [hunk 1]
```diff
@@ -578,7 +578,12 @@ ${ sqlSyntaxDropdown.sqlSyntaxDropdown() }
     </div>
 
     <div data-bind="visible: result.hasResultset() && status() == 'available' && result.data().length == 0 && result.fetchedOnce() && result.explanation().length <= 0, css: resultsKlass" style="display:none;">
-      <pre class="margin-top-10 no-margin-bottom"><i class="fa fa-check muted"></i> ${ _("Done. 0 results.") }</pre>
+      <!-- ko if: result.hasMore -->
+        <pre class="margin-top-10 no-margin-bottom"><i class="fa fa-spin fa-spinner"></i> ${ _('Loading...') }</pre>
```
Comment (romainr) [inline desktop/libs/notebook/src/notebook/templates/editor_components.mako:582]: Could we avoid the `<pre class="margin-top-10 no-margin-bottom">` duplication?

Comment (Akhilsnaik) [reply desktop/libs/notebook/src/notebook/templates/editor_components.mako:582]: Actually No , I tried that already, its been css classes have tight coupling with <pre> and <i> so the css wont work when i split it and cause height mismatch like below
So i happened not to touch the logic of seperating it.
Screenshot 2021-06-09 at 10 36 27 AM

Comment (JohanAhlen) [reply desktop/libs/notebook/src/notebook/templates/editor_components.mako:582]: You should be able to move the if inside the pre i.e.

```
<pre ...>
  <!-- ko if: result.hasMore --> 
    <i class="fa fa-spin" ....>...
  <!-- /ko -->
  <!-- ko ifnot: result.hasMore -->
    <i class="fa fa-check" ...>...
  <!-- /ko -->
</pre>
```

**Review by JohanAhlen — APPROVED** [review]

File: desktop/libs/notebook/src/notebook/templates/editor_components.mako (hunk 1 above)
Comment (JohanAhlen) [reply desktop/libs/notebook/src/notebook/templates/editor_components.mako:582]: Seems like ko comment messes with the css definition, all good with the duplication from my side.

Comment (Akhilsnaik) [reply desktop/libs/notebook/src/notebook/templates/editor_components.mako:582]: as discussed when i preceed the ko if insid pre its considering ko comment and the height of the div is pretty large causing the looks bad.
Keeping the If as same.

**Pull request merged**